internal/internal_eager.go

```diff
@@ -26,9 +26,10 @@ package internal
 type eagerWorker interface {
 	// tryReserveSlot tries to reserver a task slot on the worker without blocking
 	// caller is expected to release the slot with releaseSlot
-	tryReserveSlot() bool
+	tryReserveSlot() *SlotPermit
 	// releaseSlot release a task slot acquired by tryReserveSlot
-	releaseSlot()
+	// TODO: Make reason typed
```

Comment:
Is this TODO mean to be resolved in this PR?

Reply:
It will be. Will include it as part of adding the remaining contexts per your other comment.